sharpy/io/inout_variables.py

```diff
@@ -58,8 +58,31 @@ def get_variable_value(self, data, timestep_index=-1):
         """
         if self.node is not None:
             # structural variables for now
-            variable = getattr(data.structure.timestep_info[timestep_index], self.name)
-            if len(variable.shape) == 2:
+            try:
+                #Look for the variables in time_step_info
+                variable = getattr(data.structure.timestep_info[timestep_index], self.name)
+            except AttributeError:
+                try:
+                    #First get the dict postproc_cell and the try to find the variable in it.
+                    get_postproc_cell = getattr(data.structure.timestep_info[timestep_index], 'postproc_cell')
+                    variable = get_postproc_cell[self.name]
+                except (KeyError, AttributeError):
+                    msg = ('Node {} is neither in timestep_info nor in postproc_cell.'.format(self.node))
+                    logger.error(msg)
+                    raise IndexError(msg)
+
+            #Had to add this part for for_pos and for_vel since they are arrays.
```

Comment:
```suggestion
            # Needed for for_pos and for_vel since they are arrays.
```
Well caught, just editing the comment to make it a bit more impersonal :)